### PR TITLE
Paper: Visualization of Bioinformatics Data with Dash Bio

### PR DESCRIPTION
**Pull request by shammamah-zz**

### CODE REVIEW

**Comment by deniederhut** [comment]
Hi @shammamah ! Your reviewer (Kirtan Dave) misunderstood the review instructions and submitted their review to me via email. I'm pasting it below, and will follow up with you myself about the suggested changes. Here it is:

> Comments:
> Visualization of Bioinformatics Data with Dash Bio
> 1. The paper contains interesting Visualization of Bioinformatics Data
> 3. There are lot of scope for improvement in all part of paper to make more scientifically sound. The sample code presented here is not provided with handling of genome file format like. .bam,.sam,.gtk etc. Also, presented visualization of chromosome is only with .json file. How user can view genome with .bam and .bai file?
> 6. How the Dash Bio incorporated with other Bioinformatics data analysis pkg ?
> 7. How the visualization from Dash Bio is different from other available in Biopython?
> 8. It is not clear whether Dash Bio is cover all Bioinformatics file format and handling all kind file format for visualization. If yes so why no example presented or if no, how future improvement can be done?

> Best
> Kirtan

**Comment by shammamah-zz** [comment]
@deniederhut Thank you very much!

File: papers/shammamah_hossain/shammamah_hossain.rst [hunk 1]
```diff
@@ -0,0 +1,798 @@
...
+In the code snippet below, this parser is used to extract data from a
+PDB file that contains structural information for a small section of
+DNA: [1bna]_
+
+.. code-block:: python
+
+   >>> import json
+   >>> from dash_bio_utils import pdb_parser
+   >>> pdb_string = pdb_parser.create_data('1bna.pdb')
+   >>> 1bna = json.loads(pdb_string)
```
Comment (deniederhut) [inline papers/shammamah_hossain/shammamah_hossain.rst]: Was this supposed to be `lbna`? Python names can't start with digits.

**Review by deniederhut — CHANGES_REQUESTED** [review]
Cool! Really cool! It looks like you're missing DOIs for some of the papers you cite. Would you mind adding them in?

File: papers/shammamah_hossain/shammamah_hossain.rst (hunk 1 above)
Comment (shammamah-zz) [reply papers/shammamah_hossain/shammamah_hossain.rst]: Good catch! Fixed in ff23d98.

**Comment by shammamah-zz** [comment]
@deniederhut I've added in all DOIs here: 54cd812

I'll continue to address the comments in the review that I got over the next few days. Thank you!

**Comment by shammamah-zz** [comment]
Regarding this comment: 

>There are lot of scope for improvement in all part of paper to make more scientifically sound. 

Would it be possible to have more clarification on what exactly this means (i.e., what the paper is missing that, if added, would make it more scientifically sound)?

**Comment by deniederhut** [comment]
> i.e., what the paper is missing that, if added, would make it more scientifically sound

🤷‍♀️

So, the intention behind the way we do paper editing is that you and your reviewer have a constructive conversation, and arrive at a paper together that is clear and tightly constructed. Since Kirtan is not here to have a dialogue with you, I will say that the paper looked sound to me, although I am not a computational biologist. You are free to ignore that particular comment.

Please ping me when you feel the paper is ready for me to have a second look 😄

**Comment by shammamah-zz** [comment]
@deniederhut I believe it should be ready for a second review now! 😄

I've addressed the comments as follows: 

>There are lot of scope for improvement in all part of paper to make more scientifically sound. 

Ignoring this, as per your suggestion. 

>The sample code presented here is not provided with handling of genome file format like. .bam,.sam,.gtk etc. Also, presented visualization of chromosome is only with .json file. How user can view genome with .bam and .bai file?

This is a limitation of the package. A section on limitations and future directions has been added here: 15610ea

>How the Dash Bio incorporated with other Bioinformatics data analysis pkg ?

Again, I'm not 100% sure what this means. There is a part in the File Parsers section that mentions `GEOparse` and `parmed`, but I'm not sure if that's enough. 

>How the visualization from Dash Bio is different from other available in Biopython?

This has been addressed in 2190e5c and ae6ccbd (but again, not sure if that's enough). 

>It is not clear whether Dash Bio is cover all Bioinformatics file format and handling all kind file format for visualization. If yes so why no example presented or if no, how future improvement can be done?

This is addressed in 15610ea.

**Review by deniederhut — APPROVED** [review]
Looks good to me. Thanks for submitting to SciPy!

**Pull request merged**